kernel/main.c

```diff
@@ -200,10 +205,44 @@ read_multiboot_info(void)
       const char *name = TO_PTR(mbi->boot_loader_name);
 
       if (!strcmp(name, "TILCK_EFI")) {
+         struct tilck_extra_boot_info *extra_boot_info = TO_PTR(mbi->apm_table);
 
          printk("Multiboot: detected the TILCK_EFI bootloader\n");
-         printk("Multiboot: ACPI root ptr: %p\n", TO_PTR(mbi->apm_table));
-         acpi_set_root_pointer(mbi->apm_table);
+         printk("Multiboot: ACPI root ptr: %p\n", TO_PTR(extra_boot_info->RSDP));
+         acpi_set_root_pointer(extra_boot_info->RSDP);
+
+         /*
```

Comment:
OK, those are clearly experiments. Not OK for a merge, but it's fine to keep them while you're actively working on that.